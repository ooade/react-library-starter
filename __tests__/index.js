import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CustomButton from '../src';

configure({ adapter: new Adapter() });

describe('Library', () => {
	it('should render properly', () => {
		const wrapper = shallow(<CustomButton>Hello World</CustomButton>);
		expect(wrapper).toMatchSnapshot();
	});

	it('should match the element specified', () => {
		const actual = shallow(<CustomButton>Hello World</CustomButton>);
		const expected = (
			<button className="custom-button" type="button">
				Hello World
			</button>
		);
		expect(actual).toMatchElement(expected);
	});

	it('[Not necessary, just an example] should have className custom-button', () => {
		const wrapper = mount(<CustomButton>Hello World</CustomButton>);
		const shallowWrapper = shallow(<CustomButton>Hello World</CustomButton>);

		// mount?
		expect(wrapper.find('button')).toHaveClassName('custom-button');

		// or a simple shallow???
		expect(shallowWrapper).toHaveClassName('custom-button');
	});
});
