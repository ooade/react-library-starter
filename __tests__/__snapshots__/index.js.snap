// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library should render properly 1`] = `
<button
  className="custom-button"
  type="button"
>
  Hello World
</button>
`;
